### PR TITLE
Chimoney SlackBot 

### PR DESCRIPTION
**Pull request by thelimeskies**
### Description
This PR introduces the `/sendchimoney` command to our Slack app. This command allows users to send Chimoney to other users within the workspace. It enhances our app's functionality by facilitating payments and interactions between users.

### Testing Instructions
To test the new `/sendchimoney` command, follow these steps:

1. Join our Slack workspace for testing: [Test Workspace](https://join.slack.com/t/zee-p4j1547/shared_invite/zt-260v2mkck-EX~V4GasNEvjjH3c6aPkBw) {Recommended for now}

2. Install the app to the workspace using the following link: [Install App](https://stingray-app-3r8tj.ondigitalocean.app/slack/install)

3. Once the app is installed, open a direct message or a channel within the workspace.

4. Type `/sendchimoney` followed by the desired amount and the usernames of the recipients. For example:
   - `/sendchimoney $50 @user123 @user456` sends $50 in Chimoney to users @user123 and @user456.

5. Verify that the command executes successfully, and the recipients receive the payment link.

### Additional Notes
- Ensure that the Slack app is installed in the workspace before testing.
- If you encounter any issues during testing or have feedback, please report them in this PR.

Happy testing!
@uchibeke The slackbot

### CODE REVIEW

**Review by uchibeke — CHANGES_REQUESTED** [review]
Could you include a clear instruction on how to deploy and set this up? We should be able to follow the step by step instruction to deploy this on something like render.com or AWS so that people can install the bot.

Also, what is the difference between v2 and the existing bot?


Also, what env variables are required? What Slack Credentials are needed? Could you link the required docs, please?

**Comment by thelimeskies** [comment]
### Differences from Existing Bot (v1)

#### Introduction of the `/sendchimoney` Command
The most prominent addition in this v2 release is the `/sendchimoney` command. It allows users to send Chimoney to other users within the Slack workspace, enhancing the app's functionality by enabling quick and convenient peer-to-peer payments. This feature streamlines financial interactions and adds a new dimension to the user experience.

#### Improved Error Handling and User-Friendly Messages
v2 incorporates enhanced error handling mechanisms, resulting in more informative and user-friendly messages. Users will receive clearer notifications when errors occur, helping them understand and resolve issues more easily.

#### Enhanced Modularity
One of the most significant architectural changes in v2 is the adoption of Slack Bolt. Slack Bolt is a framework designed for building Slack apps that makes the codebase more maintainable, easily deployable, and scalable. It offers a more structured and organized approach to app development, ensuring that new features and functionality can be added with minimal effort. This modularity makes the app adaptable to changing requirements and paves the way for seamless future enhancements.

#### Multi-Workspace Support
One of the key advantages of v2 is its inherent support for multi-workspace configurations. This means that the v2 app only needs to be deployed once, and it can be effortlessly installed on any Slack workspace using the OAuth installation process. The app seamlessly adapts to different workspaces, enabling us to expand our user base across various Slack communities. Whether it's a small team or a large organization, the v2 app can be easily integrated, ensuring that users from different workspaces can enjoy its features without the need for complex modifications.


In summary, v2 represents a significant evolution of our Slack app. It introduces a powerful new payment feature, improves the user experience with clearer error handling, and leverages Slack Bolt to make the app more maintainable, easily deployable, and scalable. Moreover, the multi-workspace support offers new opportunities for growth and engagement across diverse Slack communities.

**Comment by thelimeskies** [comment]
@uchibeke The bot has been deployed already part of the feature of this v2 is it only needs to be deployed once and can installed on multiple workspaces below is a installation guide for the deployment 

would add a ReadMe with deployment guide(but having multiple deployments isn't so recommended since it's already multi workspace enabled)

Already working on getting the app published on slack app dictionary  

To install your Slack bot using the provided link, follow these steps:

1. Click on the installation link: [https://stingray-app-3r8tj.ondigitalocean.app/slack/install](https://stingray-app-3r8tj.ondigitalocean.app/slack/install).

2. You'll be redirected to the installation page. Click "Install to Workspace" to start the installation process.

3. If prompted, log in to your Slack workspace.

4. Review the requested permissions and click "Allow." This will initiate the OAuth installation process.

5. After you click "Allow," the Slack bot will be installed in your workspace.

6. To start using your newly installed bot, simply mention it in a channel or use any slash commands or interactions it supports.

Your Slack bot is now successfully installed and ready to assist in your Slack workspace.

**Comment by thelimeskies** [comment]
# Chimoney Slack Bot

The Chimoney Slack Bot is a Python application designed to facilitate money transfers within a Slack workspace. Users can send money to other users via simple commands, and the bot handles the rest.

## Features

- Send money within your Slack workspace.
- Seamless integration with Slack's UI.
- Multi-workspace support with OAuth integration.

## Installation
To install your Slack bot using the provided link, follow these steps:

Click on the installation link: https://stingray-app-3r8tj.ondigitalocean.app/slack/install.

You'll be redirected to the installation page. Click "Install to Workspace" to start the installation process.

If prompted, log in to your Slack workspace.

Review the requested permissions and click "Allow." This will initiate the OAuth installation process.

After you click "Allow," the Slack bot will be installed in your workspace.

To start using your newly installed bot, simply mention it in a channel or use any slash commands or interactions it supports.

The Slack bot is now successfully installed and ready to assist in your Slack workspace.

### Requirements

Before installing the Chimoney Slack Bot, ensure you have the following:

- Python 3.9+
- A Slack workspace where you can install the bot.
- A [Chimoney](https://chimoney.io/) account with an API key.
- A Slack Application from [Slack API](https://api.slack.com/)

### Installation Steps

1. Clone this repository to your local machine:
2. Change into the project directory:

   ```shell
   cd \submissions\Chimoney-Slackbot-V2\
   ```

3. Install project dependencies using pip:

   ```shell
   pip install -r requirements.txt
   ```

4. Create a `.env` file in the project directory and set the following environment variables:

   ```env
   SLACK_SIGNING_SECRET=your_slack_signing_secret
   CLIENT_ID=your_slack_app_client_id
   CLIENT_SECRET=your_slack_app_client_secret
   OAUTH_REDIRECT_URI=your_oauth_redirect_uri
   CHIMONEY_AUTH_KEY=your_chimoney_api_key
   ```

   Replace the placeholders with your actual credentials.


5. Start the Chimoney Slack Bot:

   ```shell
   python slackbot.py
   ```

6. The bot should now be running and listening for commands in your Slack workspace.

## Usage

To use the Chimoney Slack Bot, follow these steps:

1. Invite the bot to a channel or use a slash command.

2. Use the `/sendchimoney` command followed by the payment details. For example:

   ```
   /sendchimoney $50 @username1 @username2
   ```

3. The bot will handle the payment process and provide you with a payment link.

## Multi-Workspace Support

The Chimoney Slack Bot inherently supports multi-workspace configurations. This means that the bot can be easily deployed and used in multiple Slack workspaces without complex modifications. The bot adapts to different workspaces, enabling you to expand your user base across various Slack communities.

It only needs to be deployed once and can be installed on any workspace using OAuth.

## License

This project is licensed under the MIT License - see the [LICENSE](LICENSE) file for details.

## Acknowledgments

- [Chimoney](https://chimoney.io/) for their financial services integration.

**Comment by thelimeskies** [comment]
@uchibeke Please let me know if there's any other information you need.

Thank You

**Comment by uchibeke** [comment]
Perfect. I will approve this

Could you add the instruction guide comment to the submission readme?

**Review by uchibeke — APPROVED** [review]
